src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/SystemIPInterfaceProperties.cs

```diff
@@ -90,7 +90,7 @@ public override IPv6InterfaceProperties GetIPv6Properties()
                 throw new NetworkInformationException(SocketError.ProtocolNotSupported);
             }
 
-            return _ipv6Properties;
+            return _ipv6Properties!;
```

Comment:
What guarantees non-nullability here? Compared to the ipv4 equivalent above?

Reply:
I see that the constructor is setting that if the flag is enabled. But isn't that also the case in ipv4?

Reply:
Yes, exact same logic for ipv4, but somehow `GetIPv4Properties()` is documented to return null, discussion here: https://github.com/dotnet/runtime/pull/32990#discussion_r385942234